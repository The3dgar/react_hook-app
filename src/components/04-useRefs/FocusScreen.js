import React, { useRef } from 'react'

export const FocusScreen = () => {
  const inputRef = useRef()

  const handleClick = () => {
    inputRef.current.select()
  }
  
  return (
    <div>
      <h1>Focus</h1>
      <hr></hr>
      <input ref={inputRef} className="form-control" placeholder="Name"></input>

      <button className="btn btn-outline-primary mt-3" onClick={handleClick}>Focus</button>
    </div>
  )
}
