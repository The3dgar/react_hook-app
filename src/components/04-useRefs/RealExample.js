import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExample = () => {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>Hola ref real</h1>
      <hr></hr>
      <button className="btn btn-primary mt-5" onClick={() => setShow(!show)}>
        Show/hide
      </button>
      {show && <MultipleCustomHooks></MultipleCustomHooks>}

    </div>
  );
};
