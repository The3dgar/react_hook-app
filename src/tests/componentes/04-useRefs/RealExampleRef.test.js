import React from 'react';
import { RealExample } from '../../../components/04-useRefs/RealExample';
import { shallow } from 'enzyme';


describe("pruebas en <RealExampleRef/>", ()=> {
  const wrapper = shallow(<RealExample/>)
  test('should show <RealExampleRef/>', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false)
  })
  
  test('should show <MultipleCustomHooks/>', () => {
    wrapper.find("button").simulate("click")
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true)
    
  })
  
  
})