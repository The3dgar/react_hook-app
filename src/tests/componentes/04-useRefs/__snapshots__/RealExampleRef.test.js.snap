// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RealExampleRef/> should show <RealExampleRef/> 1`] = `
<div>
  <h1>
    Hola ref real
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
